fw/http_parser.c

```diff
@@ -2551,8 +2551,10 @@ __req_parse_accept(TfwHttpReq *req, unsigned char *data, size_t len)
 	}
 
 	__FSM_STATE(Req_I_AfterTextSlash) {
-		if (c == '*')
+		if (c == '*') {
+			__set_bit(TFW_HTTP_B_ACCEPT_HTML, req->flags);
 			__FSM_I_MOVE(I_EoT);
+		}
```

Comment:
We need a good comment in this case

Reply:
I look to this comment and also look for browsers request headers and also look at cloudfare behaviour. I as see in first task https://github.com/tempesta-tech/tempesta/issues/536:  "the limitations are required to make sure that normal browser is able to execute the challenge, e.g. if the module sent JS code in response to image request, then it'd break the application since browser doesn't try to execute execute <img>'s." We use Accept header do determine what type of content is requested and prevent challenging of img requests. As I see cloudfare challenge doesn't rely on accept header and use challenge even if Accept header is "Accept: image/*". Also I see that browsers send */* in all type of requests even if they requests images. But I don't see that browser send text/* for images and also I am not shure that browser doesn't execute code if there is an image request.

Reply:
According our investigation and discussion I drop this commit. Cloudfare has a web interface where administrator could setup rules to determine when Cloudfare should send js challenge. This rules are very wide, for example administrator can set rule, according which Cloudfare sends js challenge, when request comes from one of the chosen country! Or with header which is equal to chosen one. So We decide to create task to implement same behavior. Now we should not send js challenge to client when Accept header is */* or text/*, because browsers add */* in Accept header to all requests

Reply:
Agree, I updated https://github.com/tempesta-tech/tempesta/issues/2104#issuecomment-2063906435 on this